### ISSUE TITLE
openshift-36-metrics image missing

### ISSUE THREAD

**Issue by e-minguez** [openshift-evangelists/intro-katacoda#121]
For the using metrics scenario, the openshift-36-metrics image needs to be created.

**Comment by e-minguez** [comment]
Is there anything else I can do to speed up the image creation @BenHall ? cc @mhausenblas 
Thanks!

**Comment by BenHall** [comment]
The build is currently failing with a timeout:
```
openshift: Starting OpenShift using openshift/origin:v3.7.0 ...
    openshift: -- Checking OpenShift client ... OK
    openshift: -- Checking Docker client ... OK
    openshift: -- Checking Docker version ... OK
    openshift: -- Checking for existing OpenShift container ... OK
    openshift: -- Checking for openshift/origin:v3.7.0 image ... OK
    openshift: -- Checking Docker daemon configuration ... OK
    openshift: -- Checking for available ports ... OK
    openshift: -- Checking type of volume mount ...
    openshift:    Using nsenter mounter for OpenShift volumes
    openshift: -- Creating host directories ... OK
    openshift: -- Finding server IP ...
    openshift:    Using 127.0.0.1 as the server IP
    openshift: -- Starting OpenShift container ...
    openshift:    Creating initial OpenShift configuration
    openshift:    Starting OpenShift using container 'origin'
    openshift:    Waiting for API server to start listening
    openshift: FAIL
    openshift:    Error: timed out waiting for OpenShift container "origin"
    openshift:    WARNING: 127.0.0.1:8443 may be blocked by firewall rules
    openshift:    Details:
    openshift:      Last 10 lines of "origin" container log:
    openshift:      E0126 20:00:08.227769    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:09.853112    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:10.384951    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:11.353913    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:12.290339    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:13.300780    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:14.546959    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:15.619055    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:16.280240    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
    openshift:      E0126 20:00:17.588476    2638 controllers.go:116] Server isn't healthy yet. Waiting a little while.
```

The script is https://github.com/openshift-evangelists/intro-katacoda/tree/master/environments/openshift-3-7-clusterup

https://github.com/openshift-evangelists/intro-katacoda/blob/master/environments/openshift-3-7-clusterup/scripts/openshift.sh#L40

Any ideas why it's failing to run when inside of Packer?

**Comment by e-minguez** [comment]
Hi Ben,

The thing is, the metrics image doesn't use "oc cluster up", maybe we should follow the oc cluster up conversation in another issue. I've opened this to create the usage-metrics 3.6 image for the usage-metrics scenario -> https://github.com/openshift-evangelists/intro-katacoda/pull/109#issuecomment-358215589

Thanks.

**Comment by BenHall** [comment]
Not sure if this is an issue

```
    openshift-36-metrics: RUNNING HANDLER [openshift_metrics : restart master] ***************************
    openshift-36-metrics: fatal: [localhost]: FAILED! => {"failed": true, "msg": "The conditional check '(openshift.master.ha is not defined or not openshift.master.ha | bool) and (not (master_service_status_changed | default(false) | bool))' failed. The error was: error while evaluating conditional ((openshift.master.ha is not defined or not openshift.master.ha | bool) and (not (master_service_status_changed | default(false) | bool))): 'dict object' has no attribute 'master'\n\nThe error appears to have been in '/root/openshift-ansible-openshift-ansible-3.6.173.0.63-1/roles/openshift_metrics/handlers/main.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: restart master\n  ^ here\n"}
    openshift-36-metrics:
    openshift-36-metrics: RUNNING HANDLER [openshift_metrics : restart master api] ***********************
    openshift-36-metrics:
    openshift-36-metrics: RUNNING HANDLER [openshift_metrics : restart master controllers] ***************
    openshift-36-metrics: 	to retry, use: --limit @/root/openshift-ansible-openshift-ansible-3.6.173.0.63-1/playbooks/byo/openshift-cluster/openshift-metrics.retry
    openshift-36-metrics:
    openshift-36-metrics: PLAY RECAP *********************************************************************
    openshift-36-metrics: localhost                  : ok=177  changed=44   unreachable=0    failed=1
```

**Comment by e-minguez** [comment]
The ansible playbook last step tries to restart the master service but it fails because it doesn't exist. It is completely fine for the katacoda environment as there is no master service but the origin one.

**Comment by BenHall** [comment]
Great. The image has been built, how do we start it?

**Comment by BenHall** [comment]
ImageID `openshift-3-6-metrics` is now available on host02 of Katacoda. Once confirm the details, I'll promote to production

**Comment by e-minguez** [comment]
Thanks Ben. Once you promote it to production I will be able to do a "real world test" here https://learn.openshift.com/introduction/using-metrics/ right? Thanks!!!

**Comment by e-minguez** [comment]
Is there any way to test that image in "host02" as you said? Thanks.

**Comment by BenHall** [comment]
Update the scenario imageid to `openshift-3-6-metrics` then go via https://host02.katacoda.com/<profile> to test

**Comment by e-minguez** [comment]
Couple of things:

* I've renamed the image in the json file (it was openshift-36-metrics)
* I've added a rm to clean the /openshift.* stuff because the origin service doesn't boot trying to find a master-config.yaml which *I think* it is created by katacoda at boot, am I right?
![screenshot from 2018-01-29 16-10-37](https://user-images.githubusercontent.com/346758/35517573-25dd8fa6-050f-11e8-84a3-809164d15172.png)

**Comment by BenHall** [comment]
We do that when the service starts - https://github.com/openshift-evangelists/intro-katacoda/blob/master/environments/openshift-3-7-clusterup/scripts/openshift.sh#L64

However, we need to update the IP addresses etc. As it has a different configure script, I think we need to do some additional work here.

We'll set them as environment variables when the machine starts then the rest should call in place.

**Comment by e-minguez** [comment]
I've just added a new pull request to fix:

* Image name is now "openshfit-3-6-metrics"
* Proper deletion to be able to boot it

Please take a look and if you want to rebuild the image with those fixes, I'll test it on host02 as soon as it is available.

Thanks!
